src/main/java/edu/wpi/grip/core/Socket.java

```diff
@@ -106,6 +111,24 @@ public Step getStep() {
         return step;
     }
 
+
+    /**
+     * @param direction <code>INPUT</code> if this is the input to a step or sink, <code>OUTPUT</code> if this is the
+     *                  output of a step or source
+     */
+    public void setDirection(Direction direction) {
```

Comment:
If you are going to leave this setter here please at least remove the `public` specifier and leave it package access only.